e2e/test/testing.go

```diff
@@ -0,0 +1,69 @@
+// SPDX-FileCopyrightText: The RamenDR authors
+// SPDX-License-Identifier: Apache-2.0
+
+package test
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/go-logr/logr"
+)
+
+// T extends testing.T to use a custom logger.
+type T struct {
+	*testing.T
+	log logr.Logger
+}
+
+// WithLog returns a t wrapped with a specified log.
+// nolint: thelper
+func WithLog(t *testing.T, log logr.Logger) *T {
+	return &T{T: t, log: log}
+}
+
+// Log writes a message to the log.
+func (t *T) Log(msg string) {
+	t.log.Info(msg)
+}
+
+// Log writes a formatted message to the log.
+func (t *T) Logf(format string, args ...any) {
+	t.log.Info(fmt.Sprintf(format, args...))
+}
+
+// Error writes an error message to the log and mark the test as failed.
+func (t *T) Error(err error, msg string) {
```

Comment:
Adding error argument is not nice and require changes in current code using t.Error(err).

We can avoid it by using nil error and formatting the error into the message.
The stacktrace does not use the error, it just add it as extra field:

```
{"error": err.Error()}
```